src/components/application_manager/src/state_controller_impl.cc

```diff
@@ -955,6 +957,15 @@ void StateControllerImpl::OnStateChanged(ApplicationSharedPtr app,
 
   if (new_state->hmi_level() == mobile_apis::HMILevel::HMI_NONE) {
     app->ResetDataInNone();
+    if (mobile_apis::HMILevel::INVALID_ENUM == old_state->hmi_level()) {
+      const uint32_t app_id = app->app_id();
+      const uint32_t corr_id =
+          postponed_activation_controller_.GetPendingActivationCorrId(app_id);
+      if (corr_id > 0) {
+        app_mngr_.GetPolicyHandler().OnActivateApp(app_id, corr_id);
```

Comment:
Why is onActivateApp called before these two lines are run?
```
app_mngr_.OnHMIStateChanged(app->app_id(), old_state, new_state);
app->usage_report().RecordHmiStateChanged(new_state->hmi_level());
```

Reply:
@iCollin as I can see `OnActivateApp` indirectly triggers the same function `OnStateChanged` too.
Probably, it make sense to move these lines before

Reply:
@iCollin fixed in https://github.com/smartdevicelink/sdl_core/pull/3557/commits/4605e9859af606ef434484e8c9ea094918e4d365